tools/externaldriver/mdalDummyExternalDriver.cpp

```diff
@@ -0,0 +1,598 @@
+{
+  if ( sMeshes.find( meshId ) != sMeshes.end() )
+  {
+    const Mesh &mesh = sMeshes[meshId];
+    if ( groupIndex >= int( mesh.datasetGroups.size() ) )
+      return _return_str( "" );
+    const Datasetgroup &datasetGroup = mesh.datasetGroups.at( groupIndex );
+    return datasetGroup.metadata.at( metaDataIndex ).second.c_str();
+  }
+
+  return _return_str( "" );
+}
+
+bool CALL_CONV MDAL_DRIVER_G_datasetsDescription( int meshId, int groupIndex, bool *isScalar, int *dataLocation, int *datasetCount )
+{
+  if ( sMeshes.find( meshId ) != sMeshes.end() )
+  {
+    const Mesh &mesh = sMeshes[meshId];
+    if ( groupIndex >= int( mesh.datasetGroups.size() ) )
+      return false;
+    const Datasetgroup &datasetGroup = mesh.datasetGroups.at( groupIndex );
+    if ( !isScalar || !dataLocation || !datasetCount )
+      return false;
+    *isScalar = datasetGroup.scalar;
+    if ( datasetGroup.dataType == "onVertex" )
+      *dataLocation = 1;
+    if ( datasetGroup.dataType == "onFace" )
+      *dataLocation = 2;
+    if ( datasetGroup.dataType == "onEdge" )
+      *dataLocation = 3;
+
+    *datasetCount = datasetGroup.dataset.size();
+
+    return true;
+  }
+
+  return false;
+}
+
+int CALL_CONV MDAL_DRIVER_D_data( int meshId, int groupIndex, int datasetIndex, int indexStart, int count, double *buffer )
+{
+  if ( sMeshes.find( meshId ) != sMeshes.end() )
+  {
+    const Mesh &mesh = sMeshes[meshId];
+    if ( groupIndex >= static_cast<int>( mesh.datasetGroups.size() ) )
+      return -1;
+    const Datasetgroup &datasetGroup = mesh.datasetGroups.at( groupIndex );
+    if ( datasetIndex >= static_cast<int>( datasetGroup.dataset.size() ) )
+      return -1;
+    const Dataset &dataset = datasetGroup.dataset.at( datasetIndex );
+    int totalDatasetValues = static_cast<int>( dataset.values.size() );
+    if ( !datasetGroup.scalar )
+      totalDatasetValues /= 2;
+    int effectiveCount = std::min( count, totalDatasetValues - indexStart );
+
+    int bufferValues = effectiveCount;
+    if ( datasetGroup.scalar )
+      for ( int i = 0; i < bufferValues; ++i )
+        buffer[i] = dataset.values.at( i + indexStart );
+    else
+      for ( int i = 0; i < bufferValues * 2; ++i )
+        buffer[i] = dataset.values.at( i + indexStart * 2 );
+
+    return effectiveCount;
+  }
+
+  return -1;
+}
+
+
+bool CALL_CONV MDAL_DRIVER_D_hasActiveFlagCapability( int meshId, int groupIndex, int )
+{
+  if ( sMeshes.find( meshId ) != sMeshes.end() )
+  {
+    const Mesh &mesh = sMeshes[meshId];
+    if ( groupIndex >= static_cast<int>( mesh.datasetGroups.size() ) )
+      return false;
+    const Datasetgroup &datasetGroup = mesh.datasetGroups.at( groupIndex );
+    if ( datasetGroup.dataType == "onFace" )
+      return true;
+  }
+
+  return false;
+}
+
+//! Returns the dataset active flags
+int CALL_CONV  MDAL_DRIVER_D_activeFlags( int meshId, int groupIndex, int datasetIndex, int indexStart, int count, int *buffer )
```

Comment:
i do not think you need to repeat CALL_CONV in the implementation? also i have never seen speciying calling conventions... is it reequired at all? (looks like the one specified is default)

we do not specify it for example in the main MDAL header too

Reply:
I put this when trying to make working on windows... I tried without, it seems ok. So I will remove it.